src/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,143 @@
+  /**
+   * The class name for the root element.
+   */
+  className?: string;
+  /**
+   * The total number of steps in the progress bar.
+   * @default 1
+   */
+  totalSteps: number;
+  /**
+   * The current step in the progress bar.
+   * This should be a number between 1 and totalSteps.
+   * @default 1
+   */
+  currentStep: number;
+  /**
+   * The progress of the current step in the progress bar.
+   * This should be a number between 0 and 100.
+   * @default 0
+   * */
+  stepProgress: number;
+}
+
+function ProgressBar(
+  props: ProgressBarProps,
+  ref: Ref<HTMLDivElement>
+) {
+  const {
+    label,
+    className: classNameProp,
+    value = 0,
+    minValue = 0,
+    maxValue = 100,
+    size = "medium",
+    variant,
+    totalSteps = 1,
+    currentStep = 1,
+    stepProgress = 0,
+    isIndeterminate = false,
+    "aria-label": ariaLabel,
+    "aria-labelledby": ariaLabelledby,
+    ...rest
+  } = props;
+
+  const { progressBarProps, labelProps } = useProgressBar({ ...props, value });
+
+  const percentage = isIndeterminate
+    ? 1
+    : clamp((value - minValue) / (maxValue - minValue), 0, 1);
+  const barWidth = isIndeterminate
+    ? "100%"
+    : `${Math.round(percentage * 100)}%`;
+
+  if (!ariaLabel && !ariaLabelledby) {
+    console.warn(
+      "ProgressCircle requires an aria-label or aria-labelledby attribute for accessibility"
+    );
+  }
+
+  const StepIndicator = ({
+    index,
+    currentStep,
+    stepProgress,
+  }: {
+    index: number;
+    currentStep: number;
+    stepProgress: number;
+  }) => {
+
+    const fillStyle: CSSProperties = {
+      width: index === currentStep - 1 ? `${stepProgress}%` : "100%",
+      height: "100%",
+    };
+    const currentStepClassName = st(classes.stepIndicatorFill, {
+      isActive: index < currentStep ? true : false,
+    });
+
+    return (
+      <div key={index} className={st(classes.stepIndicator)} data-id={"stepIndicator"} >
+        <div style={fillStyle} className={currentStepClassName} data-id={"stepIndicatorFill"} />
+      </div>
+    );
+  };
+
+  return (
+    <div
+      className={st(classes.root, { isIndeterminate, size, variant, multistep: totalSteps > 1 ? true : false }, classNameProp)}
+      {...progressBarProps}
+      {...rest}
+      ref={ref}
+    >
+      <div className={st(classes.label)}>
+        {label && <span {...labelProps}>{label}</span>}
+      </div>
+      <div className={st(classes.track)}>
+        {totalSteps && totalSteps > 1 ? (
+          <div className={st(classes.stepContainer)}>
+            {Array.from({ length: totalSteps }).map((_, index) => (
+              <StepIndicator
+                key={index}
+                index={index}
+                currentStep={currentStep}
+                stepProgress={stepProgress}
+              />
+            ))}
+          </div>
+        ) : (
+          <div className={st(classes.stepIndicator)}>
+              <div
+                style={{ width: barWidth, height: 10 }}
```

Comment:
Would it be useful for `height: 10` to be part of the CSS and customisable/overridable?